poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -1185,4 +1198,303 @@ public void syncPDUID(FacilitySoftwareConfig powerIQ) {
+         }else {
+            logger.error(String.format("Invalid value for rate current : %s", rateAmps));
+         }
+      }
+      if(ratePower != null) {
+         int powerUnitIndex = ratePower.indexOf(POWERIQ_POWER_UNIT);
+         if(powerUnitIndex != -1) {
+            String powerValue = ratePower.substring(0, powerUnitIndex);
+            int rangeFlagIndex = powerValue.indexOf(RANGE_FLAG);
+            String minPowerValue = null;
+            String maxPowerValue = null;
+            if(rangeFlagIndex != -1) {
+               minPowerValue = powerValue.substring(0, rangeFlagIndex);
+               maxPowerValue = powerValue.substring(rangeFlagIndex + 1);
+               pduRateInfo.put(FlowgateConstant.PDU_MIN_RATE_POWER, minPowerValue);
+               pduRateInfo.put(FlowgateConstant.PDU_MAX_RATE_POWER, maxPowerValue);
+            }else {
+               //There is a another format of power value like 6.0kVA
+               minPowerValue = powerValue;
+               maxPowerValue = powerValue;
+               pduRateInfo.put(FlowgateConstant.PDU_MIN_RATE_POWER, minPowerValue);
+               pduRateInfo.put(FlowgateConstant.PDU_MAX_RATE_POWER, maxPowerValue);
+            }
+         }else {
+            logger.error(String.format("Invalid value for rate power : %s",ratePower));
+         }
+      }
+      if(rateVolts != null) {
+         int voltageUnitIndex = rateVolts.indexOf(POWERIQ_VOLTAGE_UNIT);
+         if(voltageUnitIndex != -1) {
+            String voltageValue = rateVolts.substring(0, voltageUnitIndex);
+            int rangeFlagIndex = voltageValue.indexOf(RANGE_FLAG);
+            String minVoltageValue = null;
+            String maxVoltageValue = null;
+            if(rangeFlagIndex != -1) {
+               minVoltageValue = voltageValue.substring(0, rangeFlagIndex);
+               maxVoltageValue = voltageValue.substring(rangeFlagIndex + 1);
+               pduRateInfo.put(FlowgateConstant.PDU_MIN_RATE_VOLTS, minVoltageValue);
+               pduRateInfo.put(FlowgateConstant.PDU_MAX_RATE_VOLTS, maxVoltageValue);
+            }else {
+               //There is a another format of voltage value like 200V
+               minVoltageValue = voltageValue;
+               maxVoltageValue = voltageValue;
+               pduRateInfo.put(FlowgateConstant.PDU_MIN_RATE_VOLTS, minVoltageValue);
+               pduRateInfo.put(FlowgateConstant.PDU_MAX_RATE_VOLTS, maxVoltageValue);
+            }
+         }else {
+            logger.error(String.format("Invalid value for rate voltage : %s",rateVolts));
+         }
+      }
+      return pduRateInfo;
+   }
+
+   private List<Asset> getAllPduAssetsFromPowerIQ(FacilitySoftwareConfig powerIQ) {
+      int limit = 100;
+      int offset = 0;
+      PowerIQAPIClient client = createClient(powerIQ);
+      List<Pdu> pdus = null;
+      List<Asset> pduAssets = new ArrayList<>();
+      Map<Long, Rack> racksMap = getRacksMap(client);
```

Comment:
Can we just query those kind of data once?   please merge the logic of sync sensor data and pdu together.

we should always try avoid to add unnecessary query pressure to the PowerIQ server.